src/Model/Request/SubModel/Head/External/Tracking.php

```diff
@@ -43,12 +41,50 @@ class Tracking extends AbstractModel
      */
     public $admittedFields = [
         'Id' => [
-            'mandatory' => false,
-            'cdata' => true,
+            'mandatory' => true,
+            'multiple' => true,
+            'instanceOf' => 'Head\\External\\Tracking\\Id',
         ],
-            'Provider' => [
-                'mandatory' => false,
-                'isAttributeTo' => 'Id',
-            ],
     ];
+
+    /**
+     * @return Id[]
+     */
+    public function getIds()
+    {
+        return $this->__get('Id');
+    }
+
+    /**
+     * @param $id
+     *
+     * @return Tracking
+     *
+     * @deprecated please use the SubModel "\RatePAY\Model\Request\SubModel\Head\External\Tracking\Id" to set the TrackingId
+     */
+    public function setId($id)
+    {
+        $this->addId((new Id())->setId($id));
+
+        return $this;
+    }
+
+    /**
+     * @param $provider
+     *
+     * @return Tracking
+     *
+     * @deprecated please use the SubModel "\RatePAY\Model\Request\SubModel\Head\External\Tracking\Id" to set the TrackingId
+     */
+    public function setProvider($provider)
+    {
+        $ids = $this->getIds();
+        if (count($ids[0]) === null) {
```

Comment:
count on `null` will throw error on PHP 7.3+

Reply:
`$ids[0]`  is also bullsh*t. :) 
fixed the whole line.

Reply:
@eiriarte-mendez changed the ubuntu-version in separate PR:
https://github.com/ratepay/php-library/pull/17
(but why? - please see my comment in the PR)